client/src/containers/Register/index.js

```diff
@@ -1,66 +1,90 @@
 import React, { PureComponent } from 'react';
 import { connect } from 'react-redux';
+
 import { getUsers, userRegister } from '../../actions';
 
+import './styles.css';
+
 class Register extends PureComponent {
     state = {
-        name: '',
-        lastname: '',
-        email: '',
-        password: '',
-        error: ''
-    }
+        formData: {
+            name: '',
+            lastname: '',
+            email: '',
+            password: ''
+        },
+        hideError: {
+            name: false,
+            lastname: false,
+            email: false,
+            password: false
+        }
+    };
 
     componentWillMount() {
         this.props.dispatch(getUsers());
     }
 
-    handleInputEmail = (event) => {
-        this.setState({ email: event.target.value });
-    } 
+    handleInput = (event) => {
+        const newFormData = { ...this.state.formData };
+        const newHideError = { ...this.state.hideError };
+        const { value, name } = event.target;
 
-    handleInputPassword= (event) => {
-        this.setState({ password: event.target.value });
-    }
+        newFormData[name] = value;
+        newHideError[name] = true;
 
-    handleInputName = (event) => {
-        this.setState({ name: event.target.value });
+        this.setState({
+            formData: newFormData,
+            hideError: newHideError
+        });
     }
 
-    handleInputLastname = (event) => {
-        this.setState({ lastname: event.target.value });
-    } 
-
     componentWillReceiveProps(nextProps) {
-        if (nextProps.users.login.isAuth) {
+        const { login } = nextProps.users;
+        if (login.isAuth) {
             this.props.history.push('/user');
-        } else if (nextProps.users.register === false) {
-            this.setState({ error: 'Error, try again' });
         } else {
             this.setState({
-                name: '',
-                lastname: '',
-                email: '',
-                password: ''
+                hideError: {
+                    name: false,
+                    lastname: false,
+                    email: false,
+                    password: false
+                }
             });
         }
     }
 
     submitForm = (event) => {
         event.preventDefault();
-        this.setState({ error: '' });
 
         this.props.dispatch(
-            userRegister({
-                email: this.state.email,
-                password: this.state.password,
-                name: this.state.name,
-                lastname: this.state.lastname
-            })
-        )
+            userRegister(this.state.formData)
+        );
+    }
+
+    getErrorClass(param) {
```

Comment:
I think that `param` should be renamed to `fieldName`.

`getErrorClass` and `renderError` functions have the same logic.
Maybe it would be good to move this condition to another helper and then use it in `getErrorClass` and `renderError`:
```
formFieldHasError(fieldName) {
        const { register } = this.props.users;
        const { hideError } = this.state;
        const errors = register && register.error && register.error.errors;

        return errors && errors[fieldName] && !hideError[fieldName];
    }
```

Reply:
Yes, good point! Thank you for the attention! 👍